File: VB/MySolution.Module.Mobile/ButtonDetailViewItemMobile.vb
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.ExpressApp.Mobile.MobileModel
Imports DevExpress.ExpressApp.Model
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace MySolution.Module.Mobile
    Public Interface IModelButtonDetailViewItemMobile
        Inherits IModelViewItem

    End Interface
    Public Class SimpleButton
        Inherits TitledComponent

        Public Sub New()
            Type = "simpleButton"
        End Sub
    End Class
    <ViewItem(GetType(IModelButtonDetailViewItemMobile))> _
    Public Class ButtonDetailViewItemMobile
        Inherits ViewItem

        Protected Overrides Function CreateControlCore() As Object
            Dim simpleButton As New SimpleButton()
            simpleButton("text") = "Click me!"
            Return simpleButton
        End Function
        Public Sub New(ByVal model As IModelViewItem, ByVal objectType As Type)
            MyBase.New(objectType, model.Id)
        End Sub
    End Class
End Namespace

File: CS/MySolution.Module.Mobile/ButtonDetailViewItemMobile.cs
using DevExpress.ExpressApp.Editors;
using DevExpress.ExpressApp.Mobile.MobileModel;
using DevExpress.ExpressApp.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MySolution.Module.Mobile {
    public interface IModelButtonDetailViewItemMobile : IModelViewItem { }
    public class SimpleButton : TitledComponent {
        public SimpleButton() {
            Type = "simpleButton";
        }
    }
    [ViewItem(typeof(IModelButtonDetailViewItemMobile))]
    public class ButtonDetailViewItemMobile : ViewItem {
        protected override object CreateControlCore() {
            SimpleButton simpleButton = new SimpleButton();
            simpleButton["text"] = "Click me!";
            return simpleButton;
        }
        public ButtonDetailViewItemMobile(IModelViewItem model, Type objectType) : base(objectType, model.Id) { }
    }
}
